internal/discovery/discovery.go

```diff
@@ -20,24 +20,31 @@ import (
 var DefaultDiscoveryInterval = time.Minute
 
 func New(
-	cc conn.Conn,
 	config config.Config,
+	grpcOptions ...grpc.DialOption,
 ) *Client {
-	c := &Client{
-		cc:      cc,
-		config:  config,
-		service: Ydb_Discovery_V1.NewDiscoveryServiceClient(cc),
+	return &Client{
+		config:      config,
+		grpcOptions: grpcOptions,
 	}
-
-	return c
 }
 
 var _ discovery.Client = &Client{}
 
 type Client struct {
-	config  config.Config
-	service Ydb_Discovery_V1.DiscoveryServiceClient
-	cc      conn.Conn
+	config      config.Config
+	grpcOptions []grpc.DialOption
+}
+
+func (c *Client) client(ctx context.Context) (Ydb_Discovery_V1.DiscoveryServiceClient, *grpc.ClientConn, error) {
+	cc, err := grpc.DialContext(ctx,
+		"dns:///"+c.config.Endpoint(),
```

Comment:
it is better to do such things once, when parsing the configuration, and not every time you connect